### PR TITLE
Reboot Policy Updates

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: wa/framework/execution.py [hunk 1]
```diff
@@ -142,7 +142,6 @@ def start_job(self):
         job_output = init_job_output(self.run_output, self.current_job)
         self.current_job.set_output(job_output)
         self.update_job_state(self.current_job)
-        self.tm.start()
```
Comment (setrofim) [inline wa/framework/execution.py]: the `stop()` should also be moved into the runner.

File: wa/framework/execution.py [hunk 2]
```diff
@@ -486,6 +486,9 @@ def run_next_job(self, context):
             if self.context.reboot_policy.reboot_on_each_job:
                 self.logger.debug('Rebooting on new job.')
                 self.context.tm.target.reboot()
+            elif self.context.reboot_policy.reboot_on_each_spec and context.spec_changed:
+                self.logger.debug('Rebooting on new spec.')
```
Comment (setrofim) [inline wa/framework/execution.py]: These messages should be upgraded to info, now that I think about it.

Comment (marcbonnici) [reply wa/framework/execution.py]: Updated and added logging for initial reboot.

File: wa/framework/execution.py (hunk 1 above)
Comment (marcbonnici) [reply wa/framework/execution.py]: Fixed.

**Pull request merged**